Cloudinary/Classes/Core/Features/Helpers/CLDTransformation.swift

```diff
@@ -1781,6 +1806,7 @@ import CoreGraphics
         case ZOOM =                         "z"
         case ASPECT_RATIO =                 "ar"
         case CUSTOM_FUNCTION =              "fn"
+        case CUSTOM_PRE_FUNCTION =          "fn_pre:"
```

Comment:
`pre` is actually part of the value (`fn` is the parameter name, the value can be with or without the `pre`).
In order for the transformation to behave correctly, the parameter key cannot repeat, and when using the key like this it's no longer guaranteed.
In other words, in this list, custom function should appear only once (as `fn`), and the value should be generated correctly with the `pre` or without it. That will also spare us from needing to deal with custom function explicitly in the string generation code (line 1749 in this class).

Reply:
@nitzanj please clarify, there can be only one custom function for a given transformation?

The current implementation is meant to allow for a situation where a transformation has both a preFunction and a custom function. Wasn't this the reason we added the pre-function?

Reply:
You can't have both in the same transformation (e.g. between slashes), it doesn't work (it's an intentional limitation in the transformation engine, not in the SDKs). You could do something like this if it's useful:
`fn_remote:aHR0c.../fn_pre:wasm:blur_wasm`

However in this case the key doesn't need to be different  since it's broken into two transformations so there's no conflict.